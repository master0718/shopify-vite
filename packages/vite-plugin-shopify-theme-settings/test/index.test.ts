import { describe, expect, it, vi } from 'vitest'
import path from 'path'
import { build } from 'vite'
import shopifyThemeSettings from '../src'
import fs from 'fs/promises'

vi.mock('../src/package.ts', () => ({
  getPackage: () => ({
    name: 'vite-plugin-shopify-theme-settings',
    version: '0.0.0',
    author: 'Barrel, LLC'
  })
}))

describe('vite-plugin-shopify', () => {
  it('builds out settings_schema.json as expected', async () => {
    await build({
      logLevel: 'silent',
      plugins: [
        shopifyThemeSettings({
          themeRoot: path.join(__dirname, '__fixtures__'),
          schemaSourceDir: path.join(__dirname, '__fixtures__', 'config', 'src')
        })
      ],
      build: {
        emptyOutDir: false,
        outDir: path.join(__dirname, '__fixtures__'),
        rollupOptions: {
          input: path.join(__dirname, '__fixtures__', 'app.js')
        }
      }
    })

    const settingsSchema = await fs.readFile(path.join(__dirname, '__fixtures__', 'config', 'settings_schema.json'), { encoding: 'utf8' })

    expect(settingsSchema).toMatchSnapshot()
  })
})
