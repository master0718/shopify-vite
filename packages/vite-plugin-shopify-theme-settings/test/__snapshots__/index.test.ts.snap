// Vitest Snapshot v1

exports[`vite-plugin-shopify > builds out settings_schema.json as expected 1`] = `
"[
  {
    \\"name\\": \\"theme_info\\",
    \\"theme_name\\": \\"vite-plugin-shopify-theme-settings\\",
    \\"theme_version\\": \\"0.0.4\\",
    \\"theme_author\\": \\"Barrel, LLC\\",
    \\"theme_documentation_url\\": \\"https://THEME_DOCUMENTATION_URL.com/\\",
    \\"theme_support_url\\": \\"https://THEME_SUPPORT_URL.com/\\"
  },
  {
    \\"name\\": \\"Group 1\\",
    \\"settings\\": [
      {
        \\"id\\": \\"group_1_title\\",
        \\"type\\": \\"text\\",
        \\"label\\": \\"Group 1 title\\"
      }
    ]
  },
  {
    \\"name\\": \\"Group 2\\",
    \\"settings\\": [
      {
        \\"id\\": \\"group_2_title\\",
        \\"type\\": \\"text\\",
        \\"label\\": \\"Group 2 title\\"
      }
    ]
  }
]
"
`;
